dbms/tests/integration/test_backup_restore/test.py

```diff
@@ -0,0 +1,151 @@
+import os.path
+import pytest
+
+from helpers.cluster import ClickHouseCluster
+from helpers.test_tools import TSV
+
+
+cluster = ClickHouseCluster(__file__)
+instance = cluster.add_instance('instance')
+q = instance.query
+path_to_data = '/var/lib/clickhouse/'
+
+
+@pytest.fixture(scope="module")
+def started_cluster():
+    try:
+        cluster.start()
+        q('CREATE DATABASE test')
+
+        yield cluster
+
+    finally:
+        cluster.shutdown()
+
+
+def exec_bash(cmd):
+    cmd = '/bin/bash -c "{}"'.format(cmd.replace('"', '\\"'))
+    return instance.exec_in_container(cmd)
+
+
+def copy_backup_to_detached(database, src_table, dst_table):
+    fp_increment = os.path.join(path_to_data, 'shadow/increment.txt')
+    increment = exec_bash('cat ' + fp_increment).strip()
+    fp_backup = os.path.join(path_to_data, 'shadow', increment, 'data', database, src_table)
+    fp_detached = os.path.join(path_to_data, 'data', database, dst_table, 'detached')
+    exec_bash('cp -r {}/* {}/'.format(fp_backup, fp_detached))
+
+
+@pytest.fixture
+def backup_restore(started_cluster):
+    q("DROP TABLE IF EXISTS test.tbl")
+    q("CREATE TABLE test.tbl (p Date, k Int8) ENGINE = MergeTree PARTITION BY toYYYYMM(p) ORDER BY p")
+    for i in range(1, 4):
+        q('INSERT INTO test.tbl (p, k) VALUES(toDate({}), {})'.format(i, i))
+    for i in range(31, 34):
+        q('INSERT INTO test.tbl (p, k) VALUES(toDate({}), {})'.format(i, i))
+
+    expected = TSV('1970-01-02\t1\n1970-01-03\t2\n1970-01-04\t3\n1970-02-01\t31\n1970-02-02\t32\n1970-02-03\t33')
+    res = q("SELECT * FROM test.tbl ORDER BY p")
+    assert(TSV(res) == expected)
+
+    q("ALTER TABLE test.tbl FREEZE")
+
+    yield
+
+    q("DROP TABLE IF EXISTS test.tbl")
+
+
+def test_restore(backup_restore):
+    q("CREATE TABLE test.tbl2 AS test.tbl")
```

Comment:
You might want to consider this: If we use a unique name for the local table in every test, it'd be trivial to make pytest run all the tests in parallel.